helix-core/src/main/java/org/apache/helix/ConfigAccessor.java

```diff
@@ -786,16 +803,25 @@ private void updateInstanceConfig(String clusterName, String instanceName,
             .forParticipant(instanceName).build();
     String zkPath = scope.getZkPath();
 
-    if (!zkClient.exists(zkPath)) {
+    if (!_zkClient.exists(zkPath)) {
       throw new HelixException(
           "updateInstanceConfig failed. Given InstanceConfig does not already exist. instance: "
               + instanceName);
     }
 
     if (overwrite) {
-      ZKUtil.createOrReplace(zkClient, zkPath, instanceConfig.getRecord(), true);
+      ZKUtil.createOrReplace(_zkClient, zkPath, instanceConfig.getRecord(), true);
     } else {
-      ZKUtil.createOrUpdate(zkClient, zkPath, instanceConfig.getRecord(), true, true);
+      ZKUtil.createOrUpdate(_zkClient, zkPath, instanceConfig.getRecord(), true, true);
+    }
+  }
+
+  /**
+   * Closes the ZkClient.
```

Comment:
minor: close the ConfigAccessor to release the ZkClient or other stateful resources.

Reply:
Done